### PR TITLE
Various profiles # 2

### PR DESCRIPTION
**Pull request by kortewegdevries**
- [mirage](https://github.com/mirukana/mirage)
- [Quaternion](https://github.com/quotient-im/Quaternion) # 1139
- [Spectral](https://gitlab.com/spectral-im/spectral)
- fractal # 1139

### CODE REVIEW

File: etc/profile-m-z/mirage.profile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+include disable-xdg.inc
+
+mkdir ${HOME}/.cache/mirage
+mkdir ${HOME}/.config/mirage
+mkdir ${HOME}/.local/share/mirage
+whitelist ${HOME}/.cache/mirage
+whitelist ${HOME}/.config/mirage
+whitelist ${HOME}/.local/share/mirage
+whitelist ${DOWNLOADS}
+whitelist /usr/share/qt
```
Comment (rusty-snake) [inline etc/profile-m-z/mirage.profile:32]: already in wusc

File: etc/profile-m-z/quaternion.profile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+include disable-programs.inc
+include disable-shell.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.cache/Quotient/quaternion
+mkdir ${HOME}/.config/Quotient
+whitelist ${HOME}/.cache/Quotient/quaternion
+whitelist ${HOME}/.config/Quotient
+whitelist ${DOWNLOADS}
+whitelist /usr/share/qt
```
Comment (rusty-snake) [inline etc/profile-m-z/quaternion.profile:26]: same

File: etc/profile-m-z/quaternion.profile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+whitelist /usr/share/Quotient/quaternion
```
Comment (rusty-snake) [inline etc/profile-m-z/quaternion.profile:27]: Is `quaternion` the only thing in `/usr/share/Quotient`?

File: etc/profile-m-z/spectral.profile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+include disable-programs.inc
+include disable-shell.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.cache/ENCOM/Spectral
+mkdir ${HOME}/.config/ENCOM
+whitelist ${HOME}/.cache/ENCOM/Spectral
+whitelist ${HOME}/.config/ENCOM
+whitelist ${DOWNLOADS}
+whitelist /usr/share/qt
```
Comment (rusty-snake) [inline etc/profile-m-z/spectral.profile:26]: same

File: etc/inc/disable-programs.inc [hunk 5]
```diff
@@ -822,6 +827,7 @@ blacklist ${HOME}/.cache/INRIA
 blacklist ${HOME}/.cache/MusicBrainz
 blacklist ${HOME}/.cache/NewsFlashGTK
 blacklist ${HOME}/.cache/QuiteRss
+blacklist ${HOME}/.cache/Quotient/quaternion
```
Comment (rusty-snake) [inline etc/inc/disable-programs.inc:830]: Would it make sense to `blacklist ${HOME}/.cache/Quotient`?

File: etc/profile-m-z/mirage.profile (hunk 1 above)
Comment (kortewegdevries) [reply etc/profile-m-z/mirage.profile:32]: Thanks I overlooked that.

File: etc/profile-m-z/quaternion.profile (hunk 3 above)
Comment (kortewegdevries) [reply etc/profile-m-z/quaternion.profile:27]: Idk if you `yay -S quaternion` and `yay -S spectral-matrix-git` they'll end up in conflicted Qt libs ending with `quotient`,you can't use both at same time without purging the other. So I think they might share.

**Comment by kortewegdevries** [comment]
Off topic: Why are many profiles missing `apparmor` even though they work with it included? There was no commit history...

**Comment by rusty-snake** [comment]
Not all contributors have a system with AA. Arch for example has no MAC in a ["regular instalation"] and Fedora uses SELinux.


["regular instalation"]: https://wiki.archlinux.org/index.php/Installation_Guide

**Comment by kortewegdevries** [comment]
> Not all contributors have a system with AA.

Oh thanks, I thought at some point it broke programs so it's removed.

File: etc/profile-m-z/quaternion.profile (hunk 3 above)
Comment (kortewegdevries) [reply etc/profile-m-z/quaternion.profile:27]: Keep this?

**Comment by kortewegdevries** [comment]
Please don't merge it right now,I'll add fractal to the list, I missed that it had a request already...

**Review by rusty-snake — COMMENTED** [review]
fractal looks good.

File: etc/profile-a-l/fractal.profile [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+tracelog
+
+disable-mnt
+private-bin fractal
+private-cache
+private-dev
+private-etc alsa,alternatives,asound.conf,ca-certificates,crypto-policies,fonts,gtk-2.0,gtk-3.0,host.conf,hostname,hosts,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,mime.types,nsswitch.conf,pki,pulse,resolv.conf,selinux,ssl,X11,xdg
+private-tmp
+
+dbus-system none
```
Comment (rusty-snake) [inline etc/profile-a-l/fractal.profile:49]: From flatpak
```
dbus-user filter
dbus-user.own org.gnome.Fractal
dbus-user.talk ca.desrt.dconf
dbus-user.talk org.freedesktop.secrets
dbus-user.talk org.freedesktop.Notifications
```

Comment (kortewegdevries) [reply etc/profile-a-l/fractal.profile:49]: I had two more using dbus-monitor something secret identity etc. Allowing this many vs. allowing all -which is better?

Comment (rusty-snake) [reply etc/profile-a-l/fractal.profile:49]: A lax filter rule is still bedder then no rules at all.

Comment (kortewegdevries) [reply etc/profile-a-l/fractal.profile:49]: I have:
```
ca.desrt.dconf.Writer
org.freedesktop.Secret.Service
org.freedesktop.Secret.Item
org.freedesktop.Secret.Collection
```

Add all?

Comment (rusty-snake) [reply etc/profile-a-l/fractal.profile:49]: talk/own rules are for _names_ (e.g. ca.desrt.dconf), ca.desrt.dconf.Writer is a _interface_.

File: etc/inc/disable-programs.inc (hunk 5 above)
Comment (kortewegdevries) [reply etc/inc/disable-programs.inc:830]: Same reason as the /usr/share/quotient one...https://github.com/quotient-im/libQuotient

**Comment by rusty-snake** [comment]
Thanks for this contribution!

**Pull request merged**